controllers/openstackdataplanedeployment_controller.go

```diff
@@ -213,13 +213,23 @@ func (r *OpenStackDataPlaneDeploymentReconciler) Reconcile(ctx context.Context,
 	// All nodeSets successfully fetched.
 	// Mark InputReadyCondition=True
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.ReadyMessage)
+	shouldRequeue := false
+	haveError := false
+
+	globalInventorySecrets := []string{}
+	globalSSHKeySecrets := map[string]string{}
+
+	// Gathering individual inventory and ssh secrets for later use
+	for _, nodeSet := range nodeSets.Items {
+		// Add inventory secret to list of inventories for global services
```

Comment:
As we're doing this in every deployment reconciliation, can we make this conditional by checking for nodesets to be deployed or service override in the deployment, containing a service with `deployOnAllNodeSets: true`?

Reply:
Also I was thinking should we make global services deployed only part of service override, as them being in nodeset spec when deployed across nodesets misleading. That brings back the discussion whether service list should even be part of nodeset spec.

Reply:
If we want to move services outside of nodeset spec, I think this goes beyond objective of this feature.
Keeping global services in separate list, as we do with Overrides, was part of earlier iteration of this feature.

Reply:
> If we want to move services outside of nodeset spec....

Neither I'm not asking to move the services out of nodeset spec in the PR (however, that's a design discussion we should have without blocking this PR), nor I want the earlier iteration of having global service list separately in nodeset spec.  The point I was making is, with service list in deployments, one can create a deployment with the whatever services to be deployed across all nodesets (without any nodesets in deployment spec or providing something like `all` ) and we deploy the services across all nodesets by collecting all nodeset inventories.

However, we should not loop over all nodesets to create list of inventories in all reconciliations as done in this PR unnecessarily. We should check if we've anything to be deployed across all nodesets before doing that.

Reply:
ServicesOverride is part of deployment spec. So effectively, your proposal would mean implementing another list in deployment spec, with list of global services. Which is what I'm talking about.

Reply:
> your proposal would mean implementing another list in deployment spec,

No that's not what I'm saying, if you read my comments properly. I'm suggesting we just use ServiceOverrides for deploy all services to start with and then move service list to deployment  spec and remove ServiceOverride.

Reply:
@rabi that would prevent to pass different config maps to a service per nodeset, e.g. nova cell config

Reply:
@gibizer If I understand correctly, you're saying we need different set of services per nodeset, as it's a different service (instance) with a different service config map. Now  nodesets in a deployment are deployed in parallel, so one can create multiple deployments, one each per cell, no? Anyway, I find the current design quite confusing where we say nodeset should have service list of services to be deployed and also have services to be deployed in all nodesets. Then we deploy all nodeset deployments multiple times in a deployment (as we do it per nodeset). Keep aside the fact that what's to be deployed in a nodeset can be overridden in a deployment.

Reply:
Could we collect the feedback about the design, with a proposal of how we'd implement that along with the impact? Part of the reason why I push back on that scope of change is that it is quite disruptive. If there was a holistic plan on how we'd implement it and mitigate the impact, perhaps it's something we could execute on.

I tend to think the NodeSet services list as the defaults for the NodeSet, which can be customized at Deployment time. I'd be in favor of building out the customization interface on the Deployment, as that is additive at this stage.

Reply:
>  Now nodesets in a deployment are deployed in parallel, so one can create multiple deployments, one each per cell, no?

Reply:
>  Now nodesets in a deployment are deployed in parallel, so one can create multiple deployments, one each per cell, no? 

then we are back to the original request to run some service against all nodes in multiple nodesets and now in multiple deplotments too.

Reply:
I agree to have a separate design discussion about this to get us on the same page.